cmd/git-go-patch/extract.go

```diff
@@ -177,18 +192,39 @@ func handleExtract(p subcmd.ParseFunc) error {
 			return fmt.Errorf("no number prefix found in %#q", path)
 		}
 		newName := fmt.Sprintf("%04v-%v", strconv.Itoa(n), after)
-
-		modifiedFile, err := os.Create(filepath.Join(tmpRenameDir, newName))
-		if err != nil {
-			return err
+		writeNewPatch := true
+
+		// Now that we're done modifying p, see if it has any effective differences vs. the old
+		// patch with the same header.
+		if matcher != nil {
+			matchingStopwatch.Start()
+			if err := matcher.Apply(path, p); err != nil {
+				return failSuggestVerbatim("failed to check patch for changes", err)
+			}
+			if matcher.LastApplyExistingMatch != "" {
```

Comment:
Rather than `LastApplyExistingMatch` being temporal state on the type, why not return the old path from `Apply`? It only seems to be used in this one place

Reply:
This is kind of based on `bufio.Scanner`--call one method to advance, and then others to check what it found afterwards. But that isn't necessarily a good pattern, and since there's only one state field and it's always used, returning it does sound better. Will change (and give it a name to fit).